### ISSUE TITLE
How to invoke notification action?

### ISSUE THREAD

**Issue by funivan** [awesomeWM/awesome#2380]
Hello. Is there any way to invoke an action on the notification.
For example, when I sea notification "New chat message" I click on it (with my mouse) and I`m in the chat (web version of the chat client)
Is there any way how to do this without a mouse. I already write code how to select specific `notification` object, but how to invoke an action on the notification.

```lua
local s = mouse.screen
for _, pos in pairs(naughty.notifications[s]) do
  for _, notification in pairs(pos) do
    -- need to invoke action on the notification  
  end
end
```

Thanks for the help.

awesome v4.2 (Human after all)
 • Compiled against Lua 5.3.1 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.4
 • LGI version: 0.9.1

**Comment by psychon** [comment]
Left click on the notification calls the `run` function that was passed in when the notification was created. `naughty.dbus` uses that for the default action of an incoming message.
To call that yourself... Well, you could dig through the internals of the notification (getting the notification wibox and simulating a mouse click on that) or you could monkey-patch `naughty.notify` and somehow "grab the stuff" you need, i.e.:
```lua
local orig = naughty.notify
local run_functions = {}
naughty.notify = function(args)
    local notification = orig(args)
    run_functions[notification] = args.run
    return notification
end
```
Then, you could simulate a left click on a notification via `run_functions[notification]()`.

Of course, this also needs some way to somehow remove old entries from this table. Turning it into a weak table (`setmetatable(run_functions, { __mode = "kv" })`) might works, but I'm not sure.

Does any of these ideas seem good to you?

**Comment by funivan** [comment]
Yes. Thank you. I can clear `run_functions` table on real run execute. 
But here is my solution:
```
 local orig = naughty.notify
    naughty.notify = function(args)
        local notification = orig(args)
        notification.run = function()
            args.run()
        end
        return notification
    end
```
Now i`m just testing this and following code works fine:
```
local n = naughty.getById(2);  -- rewriting this stuff.
n.run()
```
Is my solution ok?

**Comment by funivan** [comment]
Thanks for the help.
